### PR TITLE
Add undocumented method codecs.utf_16_be_encode.

### PR DESCRIPTION
**Pull request by rchen152**
I found the signature here:
https://github.com/python/cpython/blob/6a16b18224fa98f6d192aa5014affeccc0376eb3/Modules/_codecsmodule.c#L729

https://github.com/google/pytype/issues/348 was opened against
pytype about this method being missing.

### CODE REVIEW

File: stdlib/2and3/codecs.pyi [hunk 1]
```diff
@@ -31,6 +31,7 @@ class _IncrementalDecoder(Protocol):
 def encode(obj: _Decoded, encoding: str = ..., errors: str = ...) -> _Encoded: ...
 def decode(obj: _Encoded, encoding: str = ..., errors: str = ...) -> _Decoded: ...
 def lookup(encoding: str) -> CodecInfo: ...
+def utf_16_be_encode(obj: _Decoded, errors: str = ...) -> _Encoded: ...  # undocumented
```
Comment (JelleZijlstra) [inline stdlib/2and3/codecs.pyi]: While you're at it should we also add the corresponding decode method?

Also, both of the arguments are positional-only, so they should be prefixed with __.

Comment (rchen152) [reply stdlib/2and3/codecs.pyi]: Done and done.

File: stdlib/2and3/codecs.pyi [hunk 2]
```diff
@@ -31,7 +31,8 @@ class _IncrementalDecoder(Protocol):
 def encode(obj: _Decoded, encoding: str = ..., errors: str = ...) -> _Encoded: ...
 def decode(obj: _Encoded, encoding: str = ..., errors: str = ...) -> _Decoded: ...
 def lookup(encoding: str) -> CodecInfo: ...
-def utf_16_be_encode(obj: _Decoded, errors: str = ...) -> _Encoded: ...  # undocumented
+def utf_16_be_decode(__obj: _Encoded, __errors: str = ..., __final: bool = ...) -> _Decoded: ...  # undocumented
```
Comment (JelleZijlstra) [inline stdlib/2and3/codecs.pyi]: Both of these actually return a tuple of either decoded or encoded and an int:

```
In [117]: codecs.utf_16_be_encode('x', 'strict')                                                                                                                                                       
Out[117]: (b'\x00x', 1)

In [118]: codecs.utf_16_be_decode(b'x', 'strict')                                                                                                                                                      
Out[118]: ('', 0)
```

Comment (rchen152) [reply stdlib/2and3/codecs.pyi]: Oops - fixed, thanks.

**Comment by sproshev** [comment]
@JelleZijlstra @rchen152 Why do we continue marking positional-only parameters with `__` prefix if we have PEP 570 now?

**Comment by JelleZijlstra** [comment]
Not all type checkers support the new syntax (indeed, Python 3.8 hasn't even been released yet). For mypy in particular, there were changes to the way typed-ast is used that mean mypy will likely never support PEP 570 syntax when run on Python <3.8.

**Pull request merged**